azure-py-webserver/README.md

```diff
@@ -0,0 +1,131 @@
+1. [Configure Pulumi for Azure](https://pulumi.io/quickstart/azure/setup.html)
+1. [Configure Pulumi for Python](https://pulumi.io/reference/python.html)
+
+## Deploying and running the program
+
+1. Set up a virtual Python environment and install dependencies
+
+    ```
+    $ virtualenv -p python3 venv
+    $ source venv/bin/activate
+    $ pip install -r requirements.txt
+    ```
+
+1. Create a new stack:
+
+    ```
+    $ pulumi stack init azure-py-webserver
+    ```
+
+1. Set the Azure environment:
+
+    ```
+    $ pulumi config set azure:environment public
+    ```
+
+1. Set the required configuration for this example. This example requires you to supply a username and password to
+the virtual machine that we are going to create.
+
+    ```
+    $ pulumi config set azure-web:username myusername
+    ```
+
+    The password is a secret, so we can ask Pulumi to encrypt the configuration:
+
+    ```
+    $ pulumi config set --secret azure-web:password Hunter2hunter2
+    ```
+
+1. Run `pulumi up` to preview and deploy the changes:
+
+    ```
+    $ pulumi update
+    Previewing update (azuredev):
+
+        Type                               Name                         Plan       
+    +   pulumi:pulumi:Stack                azure-py-webserver-azuredev  create     
+    +   ├─ azure:core:ResourceGroup        server                       create     
+    +   ├─ azure:network:VirtualNetwork    server-network               create     
+    +   ├─ azure:network:PublicIp          server-ip                    create     
+    +   ├─ azure:network:Subnet            server-subnet                create     
+    +   ├─ azure:network:NetworkInterface  server-nic                   create     
+    +   └─ azure:compute:VirtualMachine    server-vm                    create     
+    
+    Resources:
+        + 7 to create
+
+    Do you want to perform this update? yes
+    Updating (azuredev):
+
+        Type                               Name                         Status      
+    +   pulumi:pulumi:Stack                azure-py-webserver-azuredev  created     
+    +   ├─ azure:core:ResourceGroup        server                       created     
+    +   ├─ azure:network:VirtualNetwork    server-network               created     
+    +   ├─ azure:network:PublicIp          server-ip                    created     
+    +   ├─ azure:network:Subnet            server-subnet                created     
+    +   ├─ azure:network:NetworkInterface  server-nic                   created     
+    +   └─ azure:compute:VirtualMachine    server-vm                    created     
+    
+    Outputs:
+        public_ip: "137.117.15.111"
+
+    Resources:
+        + 7 created
+
+    Duration: 2m55s
+
+    Permalink: https://app.pulumi.com/swgillespie/azure-py-webserver/azuredev/updates/3
+    ```
+
+1. Get the IP address of the newly-created instance from the stack's outputs: 
+
+    ```
+    $ pulumi stack output public_ip
+    137.117.15.111
```

Comment:
I feel like the other similar examples have a `curl` step here?

Reply:
This one doesn't: https://github.com/pulumi/examples/tree/master/azure-js-webserver

I suspect the other example doesn't have a curl step because it doesn't actually work. I checked out the VM instance that this example creates and I don't see the Python webserver actually running.

Reply:
Fixed in the next commit.